javascript/reactjs-todo-davinci/client/components/davinci-client/protect.js

```diff
@@ -22,10 +23,13 @@ export default function Protect({ updater, submit }) {
   useEffect(() => {
     async function handleProtect() {
       updater('fakeprofile');
-      await submit();
+      setLoading(false);
+      if (submit !== undefined) {
+        await submit();
+      }
     }
     handleProtect();
-  }, [updater, submit]);
+  }, [updater, submit, setLoading]);
```

Comment:
Man, I really wish we could improve the linting so that we didn't unnecessarily add functions like this to the dependency arrays. It's a pet peeve of mine.